homeassistant/components/goalzero/config_flow.py

```diff
@@ -20,6 +25,51 @@ class GoalZeroFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    def __init__(self):
+        """Initialize a Goal Zero Yeti flow."""
+        self.ip_address = None
+
+    async def async_step_dhcp(self, discovery_info):
+        """Handle dhcp discovery."""
+        self.ip_address = discovery_info[IP_ADDRESS]
+
+        await self.async_set_unique_id(discovery_info[MAC_ADDRESS])
+        self._abort_if_unique_id_configured(updates={CONF_HOST: self.ip_address})
+        self._async_abort_entries_match({CONF_HOST: self.ip_address})
+        try:
+            await self._async_try_connect(self.ip_address)
+        except exceptions.ConnectError:
+            return self.async_abort(reason="cannot_connect")
+        except exceptions.InvalidHost:
+            return self.async_abort(reason="invalid_host")
+        except Exception:  # pylint: disable=broad-except
+            return self.async_abort(reason="unknown")
```

Comment:
I'd log the exception stack trace at debug level. Pass `exc_info=True` in the debug log call.